doc/classes/HMACContext.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="HMACContext" inherits="Reference" version="4.0">
+	<brief_description>
+		Enables creating HMAC digests using various hashing algorithms.
+	</brief_description>
+	<description>
+		[codeblocks]
+		[gdscript]
+		extends Node
+
+		var hmac = HMACContext.new()
+
+		func _ready():
+			var key = "supersecretkey" # Key can be of any size greater than 0.
+			var data = "Return of the MAC!" # Data size can be any size greater than 0 as well.
+			hmac.start(HashingContext.HASH_SHA256, key.to_utf8())
+			hmac.update(data.to_utf8())
+			var digest = hmac.finish()
+		[/gdscript]
+		[csharp]
+		using Godot;
+
+		public class HMACExample : Node
+		{
+			public HMACContext Hmac = new HMACContext();
+			public override void _Ready()
+			{
+				string key = "supersecretkey"; // Key can be of any size greater than 0.
+				strign data = "Return of the MAC!" // Data size can be any size greater than 0 as well.
+				Hmac.start(HashType.HASH_SHA256, key.ToUTF8());
+				HMac.update(data.ToUTF8());
+				byte[] digest = Hmac.finish();
+			}
+		}
+		[/csharp]
+		[/codeblocks]
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="finish">
+			<return type="PackedByteArray">
+			</return>
+			<description>
+				Generate the HMAC digest and close the context. Once finish is called, the HMACContext should
+				not be used again.
+			</description>
+		</method>
+		<method name="start">
+			<return type="int" enum="Error">
+			</return>
+			<argument index="0" name="hash_type" type="int" enum="HashingContext.HashType">
+			</argument>
+			<argument index="1" name="key" type="PackedByteArray">
+			</argument>
+			<description>
+				Initializes the HMAC state with the HMAC key and hashing algorithm.
+			</description>
+		</method>
+		<method name="update">
+			<return type="int" enum="Error">
+			</return>
+			<argument index="0" name="data" type="PackedByteArray">
+			</argument>
+			<description>
+				Updates the HMAC message with data. Update may be called any number of times after [start] is
+				called and before [finish] is called and each subsequent call to update will append to the HMAC
+				message. This can be useful for situations where the message is being streamed over time, such
+				as from disk or from a network.
```

Comment:
```suggestion
				Updates the HMAC message with data. [method update] may be called any number of times after [method start] is called and before [method finish] is called and each subsequent call to update will append to the HMAC message. This can be useful for situations where the message is being streamed over time, such as from disk or from the network.
```